src/test/java/org/openmrs/contrib/qaframework/automation/AttachmentSteps.java

```diff
@@ -0,0 +1,120 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.AttachmentsPage;
+import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
+import org.openmrs.reference.page.FindPatientPage;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+public class AttachmentSteps extends Steps {
+
+	private AttachmentsPage attachmentsPage;
+
+	@Before(RunTest.HOOK.SELENIUM_DASHBOARD)
+	@Override
+	public void initiatePatientDashboard() {
+		initiateWithLogin();
+		findPatientPage = (FindPatientPage) homePage.goToFindPatientRecord().waitForPage();
+		findPatientPage.clickOnFirstPatient();
+		dashboardPage = (ClinicianFacingPatientDashboardPage) findPatientPage.clickOnFirstPatient().waitForPage();
+	}
+	
+	@After(RunTest.HOOK.SELENIUM_DASHBOARD)
+	public void destroy() {
+		quit();
+	}
+	
+	@Given("a user clicks on Attachments link from patient dashboard")
+	public void launchAttachmentsPage() {
+		if (dashboardPage.containsText("Start Visit")) {
+			dashboardPage.startVisit();
+			attachmentsPage = (AttachmentsPage) dashboardPage.clickOnAttachmentsAppLink().waitForPage();
+		} 
+		else if (dashboardPage.containsText("End Visit")) {
```

Comment:
refactor this part to only inculde one if to start the visit and the clicking on attatmentsPage out

Reply:
Thanks @kaweesi . This is addressed in the updated pr